docs/using-workflows/adding-triggers.md

```diff
@@ -0,0 +1,92 @@
+
+Once you've found or created a trigger container, you'll need to reference it in your workflow's `triggers` section. The `source` map should set `type: webhook` and contain an `image` field, whose value is the registry path to the container image. As with `push` triggers, the trigger definition also needs a `binding` section to map the output from entrypoint's API event onto workflow parameters.
+
+The flow of data through the system is the trickiest part of webhooks. Walking through it end-to-end:
+
+* The container entrypoint parses and extracts data from the webhook payload.
+* If the data pass validation and the workflow ought to run, the entrypoint code creates a Relay event. An event is a data structure whose field names and values Relay's service API makes available to the workflow.
+* The workflow trigger definition's `binding` section maps the content of the event to workflow parameters. The keys inside the binding are the names of the parameters and the values use `!Data <fieldname>` to extract data from the event.
+* These parameter values are then available in the `spec` section for individual steps, the same as if the workflow were run manually.
+
+When you add a webhook trigger to a workflow, the workflow's page in the Relay web app will display the automatically-generated URL for the external system to call. Adding the webhook is different for each external system, but here's an example using [this GitOps workflow](https://github.com/ahpook/relay-local/blob/main/workflows/relay-update-on-commit.yaml) to update workflows stored on the service whenever a PR merge updates them on Github.
+
+Expanding the "Settings" sidebar on the workflow's page shows the auto-generated webhook URL:
+
+![The Settings bar contains a Webhook trigger URL with a clipboard-copy helper](../images/adding-triggers-webhook-url.png)
+
+On the GitHub repository, the webhook configuration is under Settings - Webhooks. For GitHub specifically, there's the option to filter down the kinds of repository events which will call the webhook; for this workflow, only "Pull request" events make sense, so we've enabled only those events. Other services may not be as configurable, in which case the webhook entrypoint code can encode the logic to determine whether to activate the workflow. Alternately, you can use [`when` conditions](../reference/relay-workflows.md#when) in the workflow itself; this option is helpful if you're using a community-maintained webhook image that shouldn't contain your site-specific logic.
```

Comment:
It might be helpful to dig into `when` conditions in a separate section as well, since they also apply to push triggers. (And to schedule triggers, technically, although there's nothing useful to key off of.)

Reply:
i'll add a link to the longer conditionals doc since that's got the longform example https://relay.sh/docs/using-workflows/conditionals/